src/lib.rs

```diff
@@ -224,6 +224,57 @@ where
 ///     }
 /// }
 /// ```
+///
+/// # Storage modes
+///
+/// Here are the possible storage values for each query.  The default
+/// is `storage memoized`.
+///
+/// ## Input queries
+///
+/// Specifying `storage input` will give you an **input
+/// query**. Unlike derived queries, whose value is given by a
+/// function, input queries are explicit set by doing
+/// `db.query(QueryType).set(key, value)` (where `QueryType` is the
+/// `type` specified for the query).
+///
+/// If you simply specify `storage input`, you will get the default
+/// behavior:
+///
+/// - Accessing a value that has not yet been set will panic.
+/// - Each time you invoke `set`, we assume the value has changed.
+///
+/// You can change either of these two things by choosing one of the following
+/// "extended" options:
+///
+/// - `storage (input default)` -- Attempting to fetch a key that has
+///   not been explicitly `set` will use the `Default` trait to
+///   provide a value. Requires that the value implements `Default`.
+/// - `storage (input eq)` -- Compare new values against the old
+///   values and avoid triggering a new revision if they are
+///   equal. Requires value types to implement the `Eq` trait.
+/// - `storage (input default eq)` -- Both of the above.
+///
+/// ## Derived queries
+///
+/// Derived queries are specified by a function.
+///
+/// - `storage memoized` -- The result is memoized
+///   between calls.  If the inputs have changed, we will recompute
+///   the value, but then compare against the old memoized value,
+///   which can significantly reduce the amount of recomputation
+///   required in new revisions. This does require that the value
+///   implements `Eq`.
+/// - `storage volatile` -- indicates that the inputs are not fully
+///   captured by salsa. The result will be recomputed once per revision.
+/// - `storage dependencies` -- does not cache the value, so it will
+///   be recomputed every time it is needed. We do track the inputs, however,
+///   so if they have not changed, then things that rely on this query
+///   may be known not to have changed.
+/// - `storage input` -- does not cache the value, so it will
```

Comment:
What is the main difference between `storage dependencies` and `storage input` for derived queries? It seems `storage dependencies` requires a `use fn` (`dep_derived1` in _memoized_dep_inputs.rs_), but wouldn't a derived query with `storage input` just make it an input query?

Reply:
I think that's a typo maybe..?

Reply:
Er, yes, just a mistake.